docs/security/secret-detection.md

```diff
@@ -0,0 +1,111 @@
+Ensure your application data is not compromised by introducing a simple and easy way to detect secrets during its development, and gate for secrets in your CI/CD pipelines.
+Today we introduce to you an open-source secrets detection framework [Yelp Detect-Secrets](https://github.com/Yelp/detect-secrets), brought to you by Yelp.
+This tool utilizes a Shannon Entropy model to detect secrets in codebases and prevents new secrets from being entered.
+This guide will demonstrate two easy to use tools using the Detect-Secrets framework brought to you by the CSE Engineering Fundamentals and Security group to keep secrets from entering from pre-commit and pre-production environment.
+
+## Detect Secrets Pre-Commit Hook
+
+The Detect-Secrets pre-commit hook is a tool that keeps secrets from being entering the codebase at one of the earliest stages of development.
+This tool will run a detect-secrets scan right before a Git commit occurs and if any findings occur, then mitigation can occur before and history of the secret is generated in Git history.
+Using the Detect Secrets Pre-Commit approach developers can insure they don't accidentally commit secrets at the earliest point in the development cycle but at a project level it requires all developers to adopt this approach to provide protection.
+
+### Configuration for Environment
+
+Below contain two ways of setting up your environment depending on your use case.
+
+- [Pre-Commit Hook within Existing Repo](https://github.com/wbreza/pre-commit-hooks/blob/main/detect-secrets/README.md)
+  - Simple install script bootstraps existing repos with pre-commit hooks and helper scripts to help manage the secret detection lifecycle
+- [Pre-Commit Hook with New Repo Template](https://github.com/wbreza/baseline-security-seed/blob/main/SECURITY.md)
+  - Contains all of the above plus out of the box support for dev containers using the Github universal dev container image.
+  - Also includes a GitHub workflow to additionally perform secret detection within pull requests
+
+## Detect Secrets Azure DevOps Marketplace Extension
+
+This extension scans the code base for secrets by using sophisticated algorithms utilizing Shannon entropy and reports the findings into the Azure DevOps Pipeline model to enable quick feedback and response from development teams throughout the development life cycle.
+As an extension deployed in the CI/CD pipeline Detect Secrets does not rely on developers to change their development environment.
+Depending on how your pipeline is configured replying on the CI/CD extension can lead to situations where secrets are leaked into the repository before being detected.
+Leaking secrets into source the repository does require cleanup which is why using both pre-commit and CI/CD protections are recommended.
+
+### Configuration
+
+Currently this extension is in private preview and to get access reach out to a member of the Security Solution Area and we will onboard you and your crew.
```

Comment:
Small reminder that this is a public repo that is used outside of CSE as well... so we should avoid using terms like the Security Solution Area etc. Also, prefer the more generic term Team over Crew - as Crew is very CSE specific